util/stringutil/enum.go

```diff
@@ -0,0 +1,48 @@
+package stringutil
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+var titleCaser = cases.Title(language.English)
+var lowerCaser = cases.Lower(language.English)
+
+type caser interface {
+	String(s string) string
+}
+
+type Enum interface {
```

Comment:
maybe use `fmt.Stringer` instead? https://pkg.go.dev/fmt#Stringer

Reply:
So the way this works is by taking into account how an enum might look like.
Actually now that I think about it's more how a protobuf enum would look like.
maybe I should change this to:
```golang
type ProtoEnum interface {
   fmt.Stringer
}
```

Side note: this actually might be a nice protobuf plugin

Reply:
that still feels a bit odd to have it tied to the concept of an `Enum` -- the interface is simply stating that "in order to pass an object to me, it must contain a `String() string` method", nothing about that requirement is related to an Enum.

Reply:
That's true, but I'm not sure how to reconcile it with the fact the stringifying takes place according to certain expectations about the String() returned value

Reply:
is it possible to simplify your methods to just:

```
func Title(s fmt.Stringer) string {
	return pretty(titleCaser, s)
}
```

or am I missing something?

Reply:
discussed offline - agreed to change the interface to something more specific to an enum